daft/datatype.py

```diff
@@ -426,6 +426,13 @@ def infer_from_object(cls, obj: Any) -> DataType:
         """Infer Daft DataType from a Python object."""
         from daft.series import Series
 
+        try:
+            import cupy as cp
+
+            if isinstance(obj, cp.ndarray):
+                obj = obj.get()
+        except ImportError:
+            pass
         s = Series.from_pylist([obj])
```

Comment:
With the Rust changes I don't think this Python addition is needed anymore. Let's remove it.